src/scp/NominationProtocol.cpp

```diff
@@ -301,6 +371,69 @@ NominationProtocol::hashValue(Value const& value)
         mSlot.getSlotIndex(), mPreviousValue, mRoundNumber, value);
 }
 
+/*
+Used by the nomination protocol to randomize the order of messages between nodes, i.e. it tries to select a different node among the qset for each round
+of nomination.
+The terms within hasNode() to create the random value include current ledger index, 
+previous close index, round number, nodeid, and a priority boolean value.
+
+NOTE: a peer's health (it's up or down, its load, etc) is  NOT consider in computing a node's weight or priority.
+Thus you can elect a potential leader that is currently down.  When this happens, the timeout logic kicks in.
+
+@param nodeID node to copute priority
+@param node u peers we trust
+*/
+uint64 NominationProtocol::getNodePriorityWithoutWeight(NodeID const& nodeID, SCPQuorumSet const& qset) {
+
+    /* * * * * *
+    This method staysclose to original logic within getNodePrirority() to allow for easier porting to future stellar protocol versions.
+    * * * * * * */
+
+    ZoneScoped;
+    uint64 res;
+    uint64 w;
+
+    if (nodeID == mSlot.getLocalNode()->getNodeID()) //node v is the local node
```

Comment:
Maybe comment this out since we're overwriting "w" anyways, if we do want to leave the appearance of this code in to easily compare.

Reply:
The intention is to make it easier to port the code to subsequent stellar protocol thus I left it there for easy comparison.  I'll comment the block but not delete it.